packages/ui-buttons/src/Button/props.ts

```diff
@@ -29,110 +29,135 @@ import { ThemeablePropTypes } from '@instructure/emotion'
    */
-  children: PropTypes.node,
+  children?: React.ReactNode
+
   /**
    * Specifies the type of the `Button`'s underlying html element.
    */
-  type: PropTypes.oneOf(['button', 'submit', 'reset']),
+  type?: 'button' | 'submit' | 'reset'
+
   /**
    * The size of the `Button`
    */
-  size: PropTypes.oneOf(['small', 'medium', 'large']),
+  size?: 'small' | 'medium' | 'large'
+
   /**
    * Provides a reference to the `Button`'s underlying html element.
    */
-  elementRef: PropTypes.func,
+  elementRef?: (element: Element | null) => void
+
   /**
    * The element to render as the component root, `Button` by default.
    */
-  as: PropTypes.elementType,
+  as?: AsElementType
+
   /**
    * Specifies if interaction with the `Button` is enabled, disabled, or readonly.
    */
-  interaction: PropTypes.oneOf(['enabled', 'disabled', 'readonly']),
+  interaction?: 'enabled' | 'disabled' | 'readonly'
+
   /**
    * Specifies the color for the `Button`.
    */
-  color: PropTypes.oneOf([
-    'primary',
-    'primary-inverse',
-    'secondary',
-    'success',
-    'danger'
-  ]),
+  color?: 'primary' | 'primary-inverse' | 'secondary' | 'success' | 'danger'
+
   /**
    * Override the `Button`'s default focus outline color.
    */
-  focusColor: PropTypes.oneOf(['info', 'inverse']),
+  focusColor?: 'info' | 'inverse'
+
   /**
    * The `Button` display property. When set to `inline-block`, the `Button` displays inline with other elements.
    * When set to block, the `Button` expands to fill the width of the container.
    */
-  display: PropTypes.oneOf(['inline-block', 'block']),
+  display?: 'inline-block' | 'block'
+
   /**
    * Sets the alignment of the `Button` children and/or icon.
    */
-  textAlign: PropTypes.oneOf(['start', 'center']),
+  textAlign?: 'start' | 'center'
+
   /**
    * Specifies if the `Button` should render with a solid background. When false, the background is transparent.
    */
-  withBackground: PropTypes.bool,
+  withBackground?: boolean
+
   /**
    * Valid values are `0`, `none`, `auto`, `xxx-small`, `xx-small`, `x-small`,
    * `small`, `medium`, `large`, `x-large`, `xx-large`. Apply these values via
    * familiar CSS-like shorthand. For example: `margin="small auto large"`.
    */
-  margin: ThemeablePropTypes.spacing,
+  margin?: Spacing
+
   /**
    * Specify a mouse cursor to use when hovering over the button.
    * The `pointer` cursor is used by default.
    */
-  cursor: PropTypes.string,
+  cursor?: Cursor
+
   /**
    * Specifies an href attribute for the `Button`'s underlying html element.
    */
-  href: PropTypes.string,
+  href?: string
+
   /**
    * An icon, or function that returns an icon.
    */
-  renderIcon: PropTypes.oneOfType([PropTypes.node, PropTypes.func])
+  renderIcon?: React.ReactNode | (() => React.ReactNode)
+
+  /**
+   * Callback fired when the `Button` is clicked.
+   */
+  onClick?: (
+    event: React.KeyboardEvent<ViewProps> | React.MouseEvent<ViewProps>
+  ) => void
+}
```

Comment:
I explicitly added the `onClick` prop to all Buttons since they will get passed to the underlying BaseButton anyway, and this way we can pass the type of the handler.